pkg/generator/validator.go

```diff
@@ -105,3 +107,83 @@ func (v *defaultValidator) desc() *validatorDesc {
 		beforeJSONUnmarshal: false,
 	}
 }
+
+type numericValidator struct {
+	jsonName     string
+	fieldName    string
+	multipleOf   *float64
+	min          *float64
+	exclusiveMin *float64
+	max          *float64
+	exclusiveMax *float64
+}
+
+func (v *numericValidator) generate(out *codegen.Emitter) {
+	if v.multipleOf != nil {
+		v.generateMultipleOf(out)
+	}
+	if v.exclusiveMax != nil {
+		v.generateExclusiveMax(out)
+	}
+	if v.max != nil {
+		v.generateMax(out)
+	}
+	if v.exclusiveMin != nil {
+		v.generateExclusiveMin(out)
+	}
+	if v.min != nil {
+		v.generateMin(out)
+	}
+}
+
+func (v *numericValidator) desc() *validatorDesc {
+	requiredImports := make([]string, 0)
+	if v.multipleOf != nil {
+		requiredImports = append(requiredImports, "math")
+	}
+	return &validatorDesc{
+		hasError:            true,
+		beforeJSONUnmarshal: false,
+		requiredImports:     requiredImports,
+	}
+}
+
+func (v *numericValidator) generateMultipleOf(out *codegen.Emitter) {
+	out.Println(`if math.Mod(%s.%s, %f) != 0 {`, varNamePlainStruct, v.fieldName, *v.multipleOf)
```

Comment:
IIUC non-required fields are generated as pointers and this would not handle them properly (would e.g. `Mod(*float64,float64`), and no nil pointer protection). But glad to be proven wrong if that's the case! Likewise for all other `generate*` added below. I think the approach in #21 addresses this issue.